### PR TITLE
Add isa-l libraries which provides igzip program.

### PR DESCRIPTION
**Pull request by rhpvorderman**
igzip is intel's compatible implementation of gzip. It blows both `gzip` and `pigz` out of the water in terms of speed. Even on AMD processors.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by rhpvorderman** [comment]
@conda-forge/help-c-cpp I need some help on this one. I get the error: `/home/conda/staged-recipes/build_artifacts/isa-l_1593698907346/_build_env/bin/x86_64-conda_cos6-linux-gnu-as` Which supposedly does not exist. When I build locally, it does exist when I check the path but I also get the same error :confused: . As a result I have no clue how to fix this.

Also, I added some things to the meta.yml to skip Windows, yet windows still builds. I am a bit confused.

**Comment by scopatz** [comment]
`x86_64-conda_cos6-linux-gnu-as` comes from the `binutils_impl_linux-64` package

**Comment by rhpvorderman** [comment]
@scopatz I found the error! It should not use this assembler but use nasm instead. Also some of the CFLAGS passed by conda do not work with nasm.

My 'fix' was to unset the `AS` and `CFLAGS` variables in `build.sh`. I do not have a Mac or Windows, so I can not enable this package for these OS es.

**Comment by rhpvorderman** [comment]
@conda-forge/help-python-c This package is ready for review.

File: recipes/isa-l/build.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+unset AS  # This makes sure nasm is set as assembler by configure and not GCC.
+unset CFLAGS  # default CFLAGS are incompatible with nasm. Set by configure.
```
Comment (scopatz) [inline recipes/isa-l/build.sh:4]: You shouldn't unset these, but rather set them to the right values.

File: recipes/isa-l/build.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+unset AS  # This makes sure nasm is set as assembler by configure and not GCC.
```
Comment (scopatz) [inline recipes/isa-l/build.sh:3]: This could probably be `export AS="$PREFIX/bin/nasm"`

**Comment by rhpvorderman** [comment]
@scopatz  Thanks for the review. I have set the variables to the right values. Unfortunately using "$CONDA_PREFIX/bin/nasm" Does not work properly. The full path gets printed (and it exists) but the `./configure` script throws an error anyway. So I used `nasm` instead.

I also included the `igzip` name in the package description as that is probably what users will be searching for.

File: recipes/isa-l/build.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+# Use nasm as default assembler. Use relative path here as configure script
+# seems to choke on absolute paths using $CONDA_PREFIX.
+export AS=nasm
+export CFLAGS=""  # default CFLAGS are incompatible with nasm.
```
Comment (scopatz) [inline recipes/isa-l/build.sh:6]: This is probably wrong.  Can you figure out which flags are wrong and unset those?  The default flags are usually needed for proper ABI compatibility on conda-forge

Comment (rhpvorderman) [reply recipes/isa-l/build.sh]: Yes, you are right. I was being lazy and frustrated. I have checked the build again and the CFLAGS are used for both gcc and nasm, while nasm does not have necessarily all flags compatible. It is of course important that gcc uses the correct cflags.

The isa-l package does not use nasm directly, but uses a nasm-filter script to filter incompatible options for nasm. I have included a patch that removes some of the flags that conda added, were incompatible, and slipped trough the original filter.

**Comment by scopatz** [comment]
This looks great! Thanks @rhpvorderman!

**Comment by scopatz** [comment]
Let me know when you think it is ready

**Comment by rhpvorderman** [comment]
@scopatz Thanks for all your help on my first C package in conda-forge. 

I was not happy with `export AS=nasm` but this is really needed for the `./configure` script to succeed. During the `make` step, `AS` is not really used anyway, instead `./tools/nasm-filter.sh` is used. That script includes a call to `nasm`. I further modified the patch to call to `$BUILD_PREFIX/bin/nasm` instead. So now we can be certain that the `nasm` from the build environment is used and not some other `nasm`. 

I also added some tests that do some compression tasks. I figured that if the binary does not link properly to the crc and deflate libraries it might still show help correctly. Now I can be reasonably sure that every automatic update by the conda-forge bot actually works when the tests succeed.

As far as I am concerned it is ready to merge now. Thanks for your help again!

P.S. And also thanks for the nanorc repository you maintain. I love nano for editing configuration files on the terminal. Nano syntax highlighting helps a lot. Thanks!

**Comment by scopatz** [comment]
Thanks! @rhpvorderman! Also, thanks for using the nanorc files!

**Pull request merged**